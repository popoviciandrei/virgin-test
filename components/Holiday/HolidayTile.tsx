import Image from 'next/image';
import { FetchHolidaysParams, FetchHolidaysResponse, Holiday } from '@lib/api/types';

const HolidayTile = ({ holiday }: { holiday: Holiday }): JSX.Element => {
  return (
    <li>
      <h2>
        {holiday.hotel.name} {holiday.hotel.content.starRating || 1}*
      </h2>
      <p>{holiday.hotel.content.parentLocation}</p>
      <Image
        src={`https:${holiday.hotel.content.images[0].RESULTS_CAROUSEL.url}`}
        width="240"
        height="180"
        quality="75"
        alt={holiday.hotel.name}
        loading={'lazy'}
      />
      <p>
        <b>Total Price:</b> &pound;{holiday.totalPrice}
      </p>
      <p>
        <b>Hotel Facilities:</b>
      </p>
      <ul>
        {holiday.hotel.content.hotelFacilities.map((facility, index) => (
          <li key={index}>{facility}</li>
        ))}
      </ul>
    </li>
  );
};

export default HolidayTile;
