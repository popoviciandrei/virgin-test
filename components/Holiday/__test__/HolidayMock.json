{
  "totalPrice": 646.27,
  "pricePerPerson": 323.14,
  "hotel": {
    "id": "H2999",
    "name": "Sonesta ES Suites",
    "boardBasis": "Room Only",
    "content": {
      "name": "Sonesta ES Suites",
      "vRating": "4",
      "hotelDescription": "Completely reimagined, Sonesta ES Suites Orlando offers a fresh modern look throughout the recently renovated hotel. This all-suite, courtyard-style hotel is perfect for families &#x2014; with rooms that easily accommodate 6-8 people.",
      "atAGlance": [
        "Excellent location on International Drive",
        "Complimentary self-parking",
        "Complimentary WiFi",
        "Lobby lounge with daily happy hour"
      ],
      "parentLocation": "International Drive, Orlando",
      "images": [
        {
          "RESULTS_CAROUSEL": {
            "url": "//d3hk78fplavsbl.cloudfront.net/assets/common-prod/hotel/300/676023/676023-1-results_carousel.jpg?version=35"
          }
        }
      ],
      "holidayType": [
        "Families"
      ],
      "boardBasis": [
        "Bed and Breakfast"
      ],
      "hotelLocation": [
        "Close to Universal Orlando Resort"
      ],
      "accommodationType": [
        "Room"
      ],
      "hotelFacilities": [
        "Restaurant",
        "Bar",
        "Free Parking",
        "Room Service",
        "Safety Deposit Box",
        "Fitness Centre/Gym",
        "Laundry Service",
        "Internet Access",
        "Swimming Pool",
        "Hot tub"
      ],
      "starRating": "4",
      "propertyType": "Hotel"
    }
  },
  "flyingClubMiles": 1293,
  "virginPoints": 1293,
  "tierPoints": 20,
  "departureDate": "2022-05-24",
  "selectedDate": "2022-05-24"
}