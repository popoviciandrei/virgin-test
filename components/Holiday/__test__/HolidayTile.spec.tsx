import React from 'react';
import { render } from '@testing-library/react';
import HolidayMock from './HolidayMock.json';
import HolidayTile from '../HolidayTile';

describe('HolidayTile', () => {
  it('Should render the right Holiday component', () => {
    const { container } = render(<HolidayTile holiday={HolidayMock} />);
    expect(container).toMatchSnapshot();
  });
});
