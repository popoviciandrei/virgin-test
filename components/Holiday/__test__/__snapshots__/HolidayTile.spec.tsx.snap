// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HolidayTile Should render the right Holiday component 1`] = `
<div>
  <li>
    <h2>
      Sonesta ES Suites
       
      4
      *
    </h2>
    <p>
      International Drive, Orlando
    </p>
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27240%27%20height=%27180%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Sonesta ES Suites"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <p>
      <b>
        Total Price:
      </b>
       Â£
      646.27
    </p>
    <p>
      <b>
        Hotel Facilities:
      </b>
    </p>
    <ul>
      <li>
        Restaurant
      </li>
      <li>
        Bar
      </li>
      <li>
        Free Parking
      </li>
      <li>
        Room Service
      </li>
      <li>
        Safety Deposit Box
      </li>
      <li>
        Fitness Centre/Gym
      </li>
      <li>
        Laundry Service
      </li>
      <li>
        Internet Access
      </li>
      <li>
        Swimming Pool
      </li>
      <li>
        Hot tub
      </li>
    </ul>
  </li>
</div>
`;
